src/vm.rs

```diff
@@ -514,6 +569,13 @@ where
         }
     }
 
+    fn push_scalar_expression(&mut self, a: Scalar) {
```

Comment:
Instead of this helper, i'd suggest using existing `push_item` with `Expression::from(Scalar)`:

```rust
self.push_item(Expression::from(self.maxtime.into()));
```

WDTY?

Also: we are going to have ScalarKind to preserve the original in-range value (see discussion in #28). Not sure if we should try to preserve it throughout the expression, so for now we can just use Scalar.

Reply:
This is a great idea, have made the changes